RestaurantsNearMe/Presentation/ContentView.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  ContentView.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2023-08-24.
+//
+
+import SwiftUI
+import Combine
+
+struct ContentView<D: DeviceOrientationProviding>: View {
+  @ObservedObject var splashScreenViewModel: SplashScreenViewModel
+  let orientationProvider: D
```

Comment:
You might not need a generic here, did you try `let orientationProvider: some DeviceOrientationProviding`?